web/models/class/class.ex

```diff
@@ -82,4 +83,27 @@ defmodule CoursePlanner.Class do
   end
   def validate_offered_course(changeset), do: changeset
 
+  defp validate_date(%{valid?: true, changes: %{date: date} = changes} = changeset) do
+    term = OfferedCourse
+    |> Repo.get(changes[:offered_course_id] || changeset.data.offered_course_id)
```

Comment:
could you check if can use `Ecto.Changeset.get_field` instead of checking the internals of the changeset struct?